### PR TITLE
Make CI use a dummy password hasher for all tests

### PR DESCRIPTION
**Pull request by zeripath**
During the recent hash algorithm change it became clear that the choice of password hash algorithm plays a role in the time taken for CI to run.

Therefore as attempt to improve CI we should consider using a dummy hashing algorithm instead of a real hashing algorithm.

This PR creates a dummy algorithm which is then set as the default hashing algorithm during tests that use the fixtures. This hopefully will cause a reduction in the time it takes for CI to run.

### CODE REVIEW

File: modules/auth/password/hash/hash.go [hunk 1]
```diff
@@ -83,17 +83,26 @@ var (
 	availableHasherFactories = map[string]func(string) PasswordSaltHasher{}
 )
 
+// RegisterPanic registers a PasswordSaltHasher with the availableHasherFactories
+// Caution: This is not thread safe.
+func RegisterPanic[T PasswordSaltHasher](name string, newFn func(config string) T) {
```
Comment (lunny) [inline modules/auth/password/hash/hash.go]: How about the name `MustRegister`?

File: modules/auth/password/hash/dummy.go [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+	if len(salt) == 10 {
+		return string(salt) + ":" + password
+	}
+
+	return hex.EncodeToString(salt) + ":" + password
+}
+
+// NewDummyHasher is a factory method to create a DummyHasher
+// Any provided configuration is ignored
+func NewDummyHasher(config string) *DummyHasher {
```
Comment (delvh) [inline modules/auth/password/hash/dummy.go]:
```suggestion
func NewDummyHasher(_ string) *DummyHasher {
```

**Review by delvh — APPROVED** [review]
Apart from lunnys comment LGTM.

I would certainly like to know how large the speedup is, but unfortunately drone is down at the moment :frowning_face:

File: modules/auth/password/hash/hash.go (hunk 1 above)
Comment (zeripath) [reply modules/auth/password/hash/hash.go]: done

File: modules/auth/password/hash/dummy.go (hunk 2 above)
Comment (zeripath) [reply modules/auth/password/hash/dummy.go]: done

**Review by lunny — APPROVED** [review]

**Comment by delvh** [comment]
Should we really queue this PR to merge, if the CI hasn't even run once?

**Comment by lunny** [comment]
LG-TM bot

**Pull request merged**